#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 9db9af8f74be3aa4f5e630afaebe7420a7a19372 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 15 Apr 2025 16:15:15 +0200
Subject: [PATCH 3/3] Remove assert in createApplicationPalette

This was overzealous, there are consumers calling this with a null palette, and it's fine since the KColorScheme ctor handles null
---
 src/kcolorscheme.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/kcolorscheme.cpp b/src/kcolorscheme.cpp
index 1496ad6..d19bacc 100644
--- a/src/kcolorscheme.cpp
+++ b/src/kcolorscheme.cpp
@@ -657,8 +657,6 @@ bool KColorScheme::isColorSetSupported(const KSharedConfigPtr &config, KColorSch

 QPalette KColorScheme::createApplicationPalette(const KSharedConfigPtr &config)
 {
-    Q_ASSERT(config);
-
     static const QPalette::ColorGroup states[QPalette::NColorGroups] = {
         QPalette::Active, QPalette::Inactive, QPalette::Disabled
     };
--
2.49.0
